server/src/main/java/org/candlepin/auth/ConsumerAuth.java

```diff
@@ -39,41 +35,37 @@
     protected ConsumerCurator consumerCurator;
     protected OwnerCurator ownerCurator;
     protected DeletedConsumerCurator deletedConsumerCurator;
-    private Provider<I18n> i18nProvider;
 
     @Inject
     ConsumerAuth(ConsumerCurator consumerCurator, OwnerCurator ownerCurator,
-        DeletedConsumerCurator deletedConsumerCurator,
-        Provider<I18n> i18nProvider) {
+        DeletedConsumerCurator deletedConsumerCurator) {
         this.consumerCurator = consumerCurator;
         this.ownerCurator = ownerCurator;
         this.deletedConsumerCurator = deletedConsumerCurator;
-        this.i18nProvider = i18nProvider;
     }
 
+    /**
+     * Creates a principal according to the {@link Consumer} with the given
+     * consumerUuid.
+     *
+     * @param consumerUuid requested consumer
+     * @return created principal
+     */
     public ConsumerPrincipal createPrincipal(String consumerUuid) {
-        ConsumerPrincipal principal = null;
-
-        if (consumerUuid != null) {
-            // If this UUID has been deleted, return a 410.
-            if (deletedConsumerCurator.countByConsumerUuid(consumerUuid) > 0) {
-                log.debug("Key {} is deleted, throwing GoneException", consumerUuid);
-                throw new GoneException(
-                    i18nProvider.get().tr("Unit {0} has been deleted", consumerUuid), consumerUuid);
-            }
-
-            Consumer consumer = this.consumerCurator.getConsumer(consumerUuid);
-
-            if (consumer != null) {
-                Owner owner = this.ownerCurator.findOwnerById(consumer.getOwnerId());
-                principal = new ConsumerPrincipal(consumer, owner);
+        if (consumerUuid == null) {
+            return null;
+        }
 
-                if (log.isDebugEnabled() && principal != null) {
-                    log.debug("principal created for consumer {}", principal.getConsumer().getUuid());
-                }
-            }
+        final Consumer consumer = this.consumerCurator.getConsumer(consumerUuid);
+        if (consumer == null) {
+            return null;
         }
 
+        final Owner owner = this.ownerCurator.findOwnerById(consumer.getOwnerId());
+        final ConsumerPrincipal principal = new ConsumerPrincipal(consumer, owner);
+        if (log.isDebugEnabled()) {
```

Comment:
Since we're addressing this code anyway, we can drop the isDebugEnabled check here. It's already done implicitly by the .debug check, and our output code is trivial here.